packages/components/src/Calendar/CalendarRange/_docs/CalendarRange.stickersheet.stories.tsx

```diff
@@ -0,0 +1,140 @@
+import React from "react"
+import { Meta } from "@storybook/react"
+import { within } from "@storybook/testing-library"
+import {
+  StickerSheet,
+  StickerSheetStory,
+} from "~storybook/components/StickerSheet"
+import { CalendarRange, CalendarRangeProps } from "../../index"
+
+export default {
+  title: "Components/Date controls/Calendars/CalendarRange",
+  parameters: {
+    chromatic: { disable: false },
+    controls: { disable: true },
+    a11y: {
+      config: {
+        rules: [
+          {
+            // Pseudo states of Calendar days do not meet contrast ratios
+            id: "color-contrast",
+            selector:
+              ':not([data-sb-pseudo-styles="hover"], [data-sb-pseudo-styles="focus"])',
+          },
+        ],
+      },
+    },
+  },
+} satisfies Meta
+
+const CalendarRangeWrapper = (
+  props: Partial<CalendarRangeProps>
+): JSX.Element => (
+  <div data-testid={props.id}>
+    <CalendarRange
+      selected={{
+        from: new Date("2022-02-19"),
+        to: new Date("2022-03-04"),
+      }}
+      {...props}
+    />
+  </div>
+)
+
+const StickerSheetTemplate: StickerSheetStory = {
+  render: () => (
+    <>
+      <StickerSheet heading="Calendar Range">
+        <StickerSheet.Body>
+          <StickerSheet.Row rowTitle="Default">
+            <CalendarRangeWrapper />
+          </StickerSheet.Row>
+
+          <StickerSheet.Row rowTitle="With divider">
+            <div style={{ padding: "1.5rem 0" }}>
+              <CalendarRangeWrapper hasDivider />
+            </div>
+          </StickerSheet.Row>
+        </StickerSheet.Body>
+      </StickerSheet>
+
+      <StickerSheet heading="Pseudo states">
+        <StickerSheet.Header
+          headings={["Hover", "Focus"]}
+          hasVerticalHeadings
+        />
+        <StickerSheet.Body>
+          <StickerSheet.Row rowTitle="Selected (Start)">
+            <CalendarRangeWrapper id="id__calendar-range__start--hover" />
+            <CalendarRangeWrapper id="id__calendar-range__start--focus" />
+          </StickerSheet.Row>
+
+          <StickerSheet.Row rowTitle="Selected (Middle)">
+            <CalendarRangeWrapper id="id__calendar-range__middle--hover" />
+            <CalendarRangeWrapper id="id__calendar-range__middle--focus" />
+          </StickerSheet.Row>
+
+          <StickerSheet.Row rowTitle="Selected (End)">
+            <CalendarRangeWrapper id="id__calendar-range__end--hover" />
+            <CalendarRangeWrapper id="id__calendar-range__end--focus" />
+          </StickerSheet.Row>
+        </StickerSheet.Body>
+      </StickerSheet>
+    </>
+  ),
+  parameters: {
+    pseudo: {
+      hover: '[data-sb-pseudo-styles="hover"]',
+      focus: '[data-sb-pseudo-styles="focus"]',
+      focusVisible: '[data-sb-pseudo-styles="focus"]',
+    },
+  },
+  play: ({ canvasElement }): void => {
+    applyStickerSheetStyles(canvasElement)
+  },
+}
+const applyStickerSheetStyles = (canvasElement: HTMLElement): void => {
```

Comment:
This is gnarly lol

Reply:
Yeah it's pretty gross since we want to drill and target an element within elements.
Open to other suggestions if you can think of any!